osu.Game/Rulesets/Difficulty/Skills/StrainSkill.cs

```diff
@@ -60,6 +60,22 @@ public sealed override void Process(DifficultyHitObject current)
             currentSectionPeak = Math.Max(StrainValueAt(current), currentSectionPeak);
         }
 
+        /// <summary>
+        /// Adds empty strain sections
+        /// </summary>
+        public void AddEmptySections(double time, double? currentSectionOverride = null)
+        {
+            if (currentSectionOverride.HasValue)
+                currentSectionEnd = Math.Ceiling(currentSectionOverride.Value / SectionLength) * SectionLength;
+
+            while (time > currentSectionEnd)
+            {
+                saveCurrentPeak();
+                currentSectionPeak = 0; // This is wrong, but there's no way get decay from this class
```

Comment:
you can't just say "this is wrong" and then do it anyway

Reply:
Correct way to do this is applying strain decay to currentSectionPeak
but there's no way to do this under current structure of `StrainSkill`
this is possible in https://github.com/ppy/osu/pull/29290, as strain decay logic is moved back to the `StrainSkill` class